### PR TITLE
Update docs for r2c-internal-project-depends-on

### PR DESCRIPTION
**Pull request by mmcqd**
Updates the `r2c-internal-project-depends-on` docs in line with these PRs:

project-depends-on is in the rule schema and works with taint mode: https://github.com/returntocorp/semgrep/pull/4840

Findings are marked reachable/non-reachable: https://github.com/returntocorp/semgrep/pull/4908

Lockfile targeting is smart: https://github.com/returntocorp/semgrep/pull/4954

### Please ensure

- [x] A subject matter expert (SME) reviews the content
- [ ] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team
- [x] Redirects are added if the PR changes page URLs

### CODE REVIEW

**Review by ievans — APPROVED** [review]
Looks great!

File: docs/experiments/project-depends-on.md [hunk 1]
```diff
@@ -1,40 +1,72 @@
...
+## Findings
+
+Findings produced by rules with the `r2c-internal-project-depends-on` key can be of two types: _reachable_ and _nonreachable_.
+A _reachable_ finding is one with both a dependency match and a pattern match: a vulnerable dependency was found and 
+the vulnerable part of the dependency (according to the patterns in the rule) is used somewhere in code. An _unreachable_ 
+finding is one with only a dependency match. Reachable findings are reported as coming from the code that was pattern matched. 
+Unreachable findings are reported as coming from the lockfile that was dependency matched. Both kinds of findings specify their kind,
+along with all matched dependencies, in the `extra` field of semgrep's JSON output, using the `dependency_match_only` and `dependency_matches` fields, respectively.
+
+A finding will only be considered reachable if the file containing the pattern match actually depends on the dependencies in the lockfile containing the dependency match. A file depends on a lockfile if it is the nearest lockfile going up the directory tree.
```
Comment (adamkvitek) [inline docs/experiments/project-depends-on.md:68]: Change to the formatting of this section (please, delete the other lines):
```suggestion
## Findings

Findings produced by rules with the `r2c-internal-project-depends-on` rule can be of two types: _reachable_ and _nonreachable_.

- A _reachable_ finding is one with both a dependency match and a pattern match: a vulnerable dependency was found and the vulnerable part of the dependency (according to the patterns in the rule) is used somewhere in the code.
- An _unreachable_ finding is one with only a dependency match. Reachable findings are reported as coming from the code that was pattern matched. Unreachable findings are reported as coming from the lockfile that was dependency matched. Both kinds of findings specify their kind, along with all matched dependencies, in the `extra` field of Semgrep's JSON output, using the `dependency_match_only` and `dependency_matches` fields, respectively.

A finding is only considered reachable if the file containing the pattern match actually depends on the dependencies in the lockfile containing the dependency match. A file depends on a lockfile if it is the nearest lockfile going up the directory tree.
```

Comment (adamkvitek) [reply docs/experiments/project-depends-on.md:68]: On top of formatting suggestion, I have some ideas for the unreachable definition (I am not sure about the "Both kinds of findings specify their kind" - which kind? See my suggestion below, I hope it makes sense):

Original:
- An _unreachable_ finding is one with only a dependency match. Reachable findings are reported as coming from the code that was pattern matched. Unreachable findings are reported as coming from the lockfile that was dependency matched. Both kinds of findings specify their kind, along with all matched dependencies, in the `extra` field of Semgrep's JSON output, using the `dependency_match_only` and `dependency_matches` fields, respectively.

Suggesting:
- An _unreachable_ finding is one with only a dependency match. Reachable findings are reported as coming from the **code** that was pattern matched. Unreachable findings are reported as coming from the **lockfile** that was dependency matched. For both types of findings, Semgrep specifies whether they are unreachable or reachable along with all matched dependencies, in the `extra` field of Semgrep's JSON output, using the `dependency_match_only` and `dependency_matches` fields, respectively.

**Review by adamkvitek — COMMENTED** [review]
Review done @mmcqd, let me know when you implement it and I will have a quick look again. Thank you for your great contribution!

I was not sure about the use of words rule versus key in this article. See discussion: https://returntocorp.slack.com/archives/C02JT6YA5S7/p1649436380472059?thread_ts=1649429478.115609&cid=C02JT6YA5S7

File: docs/experiments/project-depends-on.md [hunk 2]
```diff
@@ -1,40 +1,72 @@
...
+```yaml
+r2c-internal-project-depends-on:
+  depends-on-either:
+    - namespace: ...
+      package: ...
+      version: ...
+    - namespace: ...
+      package: ...
+      version: ...
+    ...
```
[the remark this answers is not in the document]
Comment (mmcqd) [reply docs/experiments/project-depends-on.md:40]: The `...` at the end are to indicate you could have more dependencies (as many as you want)

**Pull request merged**